### ISSUE TITLE
Behaviour of Type[Any]

### ISSUE THREAD

**Issue by dmoisset** [python/mypy#3098]
Currently a `Type[Any]` is considered to have the same attributes as `Type[object]`. However, actually it probably has many attributes that could be used, especially when the `Any` comes from a module without stubs. Consider the following example:

```python
from typing import Type
from some_module import Candy, Lollipop  # some_module has no stubs/annotations

class CandyFactory:
    manufactured_candy: Type[Candy]

cf = CandyFactory()
cf.manufactured_candy = Lollipop

print(cf.manufactured_candy.sugar_level)
```

Then running:

```
$ mypy --ignore-missing-imports candy.py 
candy.py:11: error: Type[Any] has no attribute "sugar_level"
```
Which seems to go a bit against the idea of gradual typing and intention of `Any`.

One possible solution is to consider that `Type[Any] == Any`, I think this actually would not miss anything that could be catched now, while reducing false positives.

**Comment by JukkaL** [comment]
I've wondered myself about how `Type[Any]` should behave, and your reasoning sounds about right. It should probably not be the same as `Any` with respect to subtyping/compatibility, only with respect to attribute access. Passing `Type[Any]` to a function that excepts an `int` should still be rejected (and vice versa).

**Comment by JukkaL** [comment]
I'm spelling out explicitly what should change: 

For a value `x` of type `Type[Any]`, mypy should allow arbitrary attribute access like `x.foobar`, and the result type should be `Any`.
